imod/mf6/auxiliary_variables.py

```diff
@@ -25,11 +25,40 @@ def _get_auxiliary_data_variable_names_mapping(package):
     return result
 
 
-def add_periodic_auxiliary_variable(package):
+def add_periodic_auxiliary_variable(package) -> None:
+    """
+    splits an auxiliary dataarray (with one or more auxiliary variable dimension) into dataarrays per
+    auxiliary variable dimension. For example a concentration auxiliary variable in a flow package
+    will have a species dimension, and will be split in several dataarrays- one for each species.
+    """
     if hasattr(package, "_auxiliary_data"):
         for aux_var_name, aux_var_dimensions in package._auxiliary_data.items():
             aux_coords = package.dataset[aux_var_name].coords[aux_var_dimensions].values
             for s in aux_coords:
                 package.dataset[s] = package.dataset[aux_var_name].sel(
                     {aux_var_dimensions: s}
                 )
+
+
+def remove_periodic_auxiliary_variable(package) -> None:
```

Comment:
What about the following method names?
```
add_periodic_auxiliary_variable -> add_periodic_auxiliary_variable_to_dataset
remove_periodic_auxiliary_variable -> remove_periodic_auxiliary_variable_from_dataset
has_auxiliary_variable_source_array -> has_auxiliary_variable_in_data_set
```

Anyhow the term `source array `is confusing. I think `dataset` or `package dataset`  is clearer

Reply:
renamed them. I removed the "periodic"part because these auxiliary variables contain data that is potentially time dependent, but usually not periodic. 
I now call them 
add_periodic_auxiliary_variable -> expand_auxiliary_variables
remove_periodic_auxiliary_variable ->remove_expanded_auxiliary_variables_from_dataset
has_auxiliary_variable_source_array -> removed